go-chaos/cmd/worker.go

```diff
@@ -0,0 +1,112 @@
+// Copyright 2022 Camunda Services GmbH
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"context"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/camunda/zeebe/clients/go/v8/pkg/entities"
+	"github.com/camunda/zeebe/clients/go/v8/pkg/worker"
+	"github.com/camunda/zeebe/clients/go/v8/pkg/zbc"
+	"github.com/spf13/cobra"
+)
+
+const jobType = "zbchaos"
+
+func init() {
+	rootCmd.AddCommand(workerCommand)
+}
+
+var workerCommand = &cobra.Command{
+	Use:   "worker",
+	Short: "Starts a worker for zbchaos jobs",
+	Long:  "Starts a worker for zbchaos jobs that executes zbchaos commands",
+	Run:   start_worker,
+}
+
+type ChaosProvider struct {
+	path    string
+	args    []string
+	timeout int64
+}
+
+type ZbChaosVariables struct {
+	clusterId string
+	provider  ChaosProvider
+}
+
+func start_worker(cmd *cobra.Command, args []string) {
+	credsProvider, err := zbc.NewOAuthCredentialsProvider(&zbc.OAuthProviderConfig{
+		ClientID:               os.Getenv("TESTBENCH_CLIENT_ID"),
+		ClientSecret:           os.Getenv("TESTBENCH_CLIENT_SECRET"),
+		Audience:               strings.TrimSuffix(os.Getenv("TESTBENCH_ADDRESS"), ":443"),
+		AuthorizationServerURL: os.Getenv("TESTBENCH_AUTHORIZATION_SERVER_URL"),
```

Comment:
:x: Would be great if we could leave testbench out here. Can you check whether the go client already supports env vars for the cloud client? Maybe this is not event necessary.

The testbench env vars can then be injected into the docker image for example